app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,77 @@
+# frozen_string_literal: true
+
+class PostsController < ApplicationController
+  before_action :set_post, only: %i[show edit update destroy]
+
+  # GET /posts or /posts.json
+  def index
+    @posts = Post.all
+  end
+
+  # GET /posts/1 or /posts/1.json
+  def show
+    content_not_found if @post.blank?
+    @comment = @post.comments.all
+  end
+
+  # GET /posts/new
+  def new
+    @post = Post.new
+  end
+
+  # GET /posts/1/edit
+  def edit
+    content_not_found if current_user.id != @post.user_id
```

Comment:
Handling this with edit and ignoring update and other actions will be a huge risk as stated above. Use Pundit.